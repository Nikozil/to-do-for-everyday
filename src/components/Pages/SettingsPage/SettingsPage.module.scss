.wrapper {
  display: flex;
  align-items: center;
  justify-content: center;
  height: 100%;
}
.content {
  display: flex;
  flex-direction: column;
  justify-content: space-between;

  height: 95%;
  width: 100%;
  min-width: 200px;
  max-width: 400px;
  margin: 0 5px 0 5px;
  text-align: center;
}
