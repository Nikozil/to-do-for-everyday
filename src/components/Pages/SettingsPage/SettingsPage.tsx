import React, { MouseEvent } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import {
  signOut,
  updatePassword,
  updateProfile,
} from '../../../Redux/modules/userSlice';
import { selectUser } from '../../../Redux/selectors/userSelector';
import ChangeNameForm from '../../Forms/ChangeNameForm/ChangeNameForm';
import ChangePasswordForm from '../../Forms/ChangePasswordForm/ChangePasswordForm';
import styles from './SettingsPage.module.scss';

const SettingsPage = () => {
  const dispatch = useDispatch();

  const user = useSelector(selectUser);

  const ChangeNameFormHandleSubmit = async (displayName: string) => {
    await dispatch(updateProfile(displayName));
  };

  const ChangePasswordFormHandleSubmit = async (
    oldPassword: string,
    newPassword: string
  ) => {
    const response = await dispatch(updatePassword(oldPassword, newPassword));
    return response;
  };

  const signoutHandler = (event: MouseEvent<HTMLButtonElement>) => {
    event.preventDefault();
    dispatch(signOut());
  };

  return (
    <div className={styles.settings}>
      <div className={styles['settings__content']}>
        <ChangeNameForm
          displayName={user.userData.displayName}
          handleSubmit={ChangeNameFormHandleSubmit}
        />
        <ChangePasswordForm handleSubmit={ChangePasswordFormHandleSubmit} />
        <button className={'btn btn-secondary  my-5'} onClick={signoutHandler}>
          Выйти
        </button>
      </div>
    </div>
  );
};

export default SettingsPage;
