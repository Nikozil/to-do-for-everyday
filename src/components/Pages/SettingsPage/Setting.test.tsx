import { render } from '../../../utils/tests/test-utils';
import SettingsPage from './SettingsPage';

describe('Settings Page tests', () => {
  it('render correct', () => {
    const { asFragment } = render(<SettingsPage />);
    expect(asFragment()).toMatchSnapshot();
  });
});
