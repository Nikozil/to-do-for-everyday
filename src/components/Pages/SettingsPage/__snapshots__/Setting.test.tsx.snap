// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings Page tests render correct 1`] = `
<DocumentFragment>
  <div
    class="settings"
  >
    <div
      class="settings__content"
    >
      <form
        action="#"
      >
        <div
          class="mb-1"
        >
          <label
            class="form-label"
            for="displayName"
          >
            Имя
          </label>
          <input
            class="form-control"
            id="displayName"
            name="displayName"
            placeholder="Имя"
            type="text"
            value=""
          />
        </div>
        <div
          class="text-center my-2"
        >
          <button
            class="btn btn-success center-block"
            type="submit"
          >
            Изменить
          </button>
        </div>
      </form>
      <form
        action="#"
      >
        <div
          class="mb-1"
        >
          <label
            class="form-label"
            for="oldPassword"
          >
            Старый пароль
          </label>
          <input
            class="form-control"
            id="oldPassword"
            name="oldPassword"
            placeholder=""
            type="password"
            value=""
          />
        </div>
        <div
          class="form-text"
          style="height: 21px;"
        />
        <div
          class="mb-1"
        >
          <label
            class="form-label"
            for="newPassword"
          >
            Новый пароль
          </label>
          <input
            class="form-control"
            id="newPassword"
            name="newPassword"
            placeholder=""
            type="password"
            value=""
          />
        </div>
        <div
          class="form-text"
          style="height: 21px;"
        />
        <div
          class="mb-1"
        >
          <label
            class="form-label"
            for="confirmNewPassword"
          >
            Подтвердите новый пароль
          </label>
          <input
            class="form-control"
            id="confirmNewPassword"
            name="confirmNewPassword"
            placeholder=""
            type="password"
            value=""
          />
        </div>
        <div
          class="form-text"
          style="height: 21px;"
        />
        <div
          class="text-center my-2"
        >
          <button
            class="btn btn-success center-block"
            type="submit"
          >
            Изменить
          </button>
        </div>
      </form>
      <button
        class="btn btn-secondary  my-5"
      >
        Выйти
      </button>
    </div>
  </div>
   
</DocumentFragment>
`;
